File: admin/update_status.py
from flask import Blueprint, request, jsonify
import mysql.connector
import os

admin_bp = Blueprint('admin', __name__)

def get_db_connection():
    return mysql.connector.connect(
        host=os.getenv("DB_HOST"),
        user=os.getenv("DB_USER"),
        password=os.getenv("DB_PASS"),
        database=os.getenv("DB_NAME")
    )

@admin_bp.route('/admin/update-status', methods=['POST'])
def update_status():
    data = request.get_json()
    status = data.get('status')
    user_id = data.get('id')

    if not status or not user_id:
        return jsonify({"error": "Missing status or id"}), 400

    conn = get_db_connection()
    cursor = conn.cursor()

    try:
        sql = "UPDATE users SET status = %s WHERE id = %s"
        cursor.execute(sql, (status, user_id))
        conn.commit()
        return jsonify({"success": True})
    except Exception as e:
        return jsonify({"error": str(e)}), 500
    finally:
        cursor.close()
        conn.close()

File: admin/update-status.php
<?php
if ($_SERVER["REQUEST_METHOD"] === "POST") {
    include '../db.php';  // adjust path if your db.php is elsewhere

    $email = $_POST["email"];
    $status = $_POST["status"];

    // Prepare statement to avoid SQL injection
    $stmt = $conn->prepare("UPDATE users SET status=? WHERE email=?");
    $stmt->bind_param("ss", $status, $email);

    if ($stmt->execute()) {
        echo "✅ User status updated to $status.";
    } else {
        echo "❌ Error: " . $stmt->error;
    }

    $stmt->close();
    $conn->close();
}
?>

<h2>Update User Status</h2>
<form method="POST">
  <input type="email" name="email" placeholder="User Email" required><br><br>
  <select name="status" required>
    <option value="paid">Paid</option>
    <option value="unpaid">Unpaid</option>
  </select><br><br>
  <button type="submit">Update Status</button>
</form>
